telemetry/telemetrybuffer.go

```diff
@@ -216,7 +216,12 @@ func (tb *TelemetryBuffer) Write(b []byte) (c int, err error) {
 
 // Cancel - signal to tear down telemetry buffer
 func (tb *TelemetryBuffer) Cancel() {
-	tb.cancel <- true
```

Comment:
I have not read through how the value sent on this channel is being used, so I don't know how messy it is. But the better pattern for this kind of binary "write once, read any" signal via a channel is to `close` the channel instead of sending a single item through it. Closing is non-blocking, and any reader waiting on the channel will immediately unblock and read a zero-value from the channel, and, if using two-value reads (`v, ok := <-tb.cancel`), a [`false` indicating that the channel is closed](https://go.dev/ref/spec#Receive_operator).
It's important that this be the only writer for the channel, because any subsequent attempt to send on the closed channel will panic.